### PR TITLE
deps!: bump to celestia-app `v0.15.0-rc0`

### PR DESCRIPTION
**Pull request by rootulp**
Bump to celestia-app https://github.com/celestiaorg/celestia-app/releases/tag/v0.15.0-rc0 which contains two notable changes to namespaces:
1. Increase the namespace size from 8 bytes to 32 bytes. See [ADR-015](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-015-namespace-id-size.md)
2. Prefix the namespace with a version byte. See [ADR-014](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-014-versioned-namespaces.md)

So a namespace is now `namespace version (1 byte) | namespace ID (32 bytes)`. See [specs](https://celestiaorg.github.io/celestia-app/specs/data_structures.html#namespace) for more info.

FYI celestia-app `v0.15.0-rc0` won't be deployed to BSR so maintainers should decide when it is appropriate to merge this PR.

### CODE REVIEW

File: share/ipld/nmt.go
Comment (Wondertan) [inline share/ipld/nmt.go:39]: Yep. Let's just remove 8

File: share/getter.go
Comment (rootulp) [inline share/getter.go:68]: @Wondertan @evan-forbes what are your thoughts on this approach? The previous implementation depended on the fact that the row roots were constructed with the property: `isMaxNamespace=true` so that it could efficiently generate a list of `originalRoots` and then use those roots to validate the NMT proofs of each NamespacedRow.

With the new approach, we remove the dependency on the `isMaxNamespace=true` property at the computational cost of attempting to validate one NMT proof per row root in the DAH.

Comment (rootulp) [reply share/getter.go:68]: An alternative is to change how parity namespace is constructed in celestia-app so that we can still use the `ignoreMaxNamespace=true` feature of the NMT. I'll explore that now in case there are concerns on computational complexity of attempting verify NMT proofs w/ multiple roots

File: share/eds/eds_test.go
Comment (rootulp) [inline share/eds/eds_test.go:47]: Go was using the last value appended to `doubleNID` and `parity` for all slice values so `want` looked incorrect:

```
        	            	expected: [][]uint8{[]uint8{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4}, []uint8{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4}, []uint8{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4}, []uint8{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}, []uint8{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x45}}
```

Notice the `4` suffix for the first 4 byte slices. Remediated by inlining the `append`s

File: share/getters/getter_test.go
Comment (rootulp) [inline share/getters/getter_test.go:142]: this comment was misleading

File: share/getter.go
Comment (rootulp) [reply share/getter.go]: opted to not go with this approach so resolving.

File: go.mod
Comment (rootulp) [inline go.mod:62]: this bump wasn't intentional and appears to have been auto applied by `go mod tidy` (?)

File: share/eds/testdata/example-root.json
Comment (rootulp) [inline share/eds/testdata/example-root.json:3]: the example had to be updated b/c row_roots and col_roots are now longer.

Before: 2 * 8 bytes (namespace) + 32 bytes = 48 bytes
After: 2 * 33 bytes (namespace) + 32 bytes = 98 bytes

**Comment by rootulp** [comment]
https://github.com/celestiaorg/celestia-node/actions/runs/4694062681/jobs/8321808717?pr=1999#step:4:609 there's a panic in NMT which we should replace with an error. Ref: https://github.com/celestiaorg/rsmt2d/issues/156

**Comment by rootulp** [comment]
https://github.com/celestiaorg/celestia-node/actions/runs/4700412090/jobs/8335054478#step:4:563 is failing but I assume that's not blocking this PR because it also occurred on https://github.com/celestiaorg/celestia-node/pull/2079

File: share/empty.go
Comment (Wondertan) [inline share/empty.go:74]: If we panic above anyway and sure that this func always succeeds with given input, we can just panic here avoiding error return as it was

**Comment by musalbas** [comment]
I think we should also modify the celestia-node RPC so that it rejects PFBs with invalid namespaces that will be rejected by the consensus network? eg invalid namespace version, or does not have 22 leading zeros.

File: go.mod
Comment (Wondertan) [reply go.mod:63]: Not a problem.

File: share/availability/light/availability_test.go
Comment (Wondertan) [inline share/availability/light/availability_test.go:19]: It seems like we would need to depend on this type everywhere rather than on low-level namespace at some point. I will likely tackle that during share pkg extraction. It will require some refactorings anyway

File: share/eds/eds.go
Comment (Wondertan) [inline share/eds/eds.go:226]: I looked at the Bytes implementation, and it allocates 😕 
This method will be called on the hottest code path in the node and very likely in the core.
The namespace type should have been a singular field type where we would pick the version and all the other details by directly accessing the subslice through respecting methods.

**Review by Wondertan — APPROVED** [review]
So we were not using the constants everywhere and had 8 hardcoded in tests. Thank you for finding all of them!

We should decide how and then merge this breaking PR. Likely when we will go through the upgrade process during ITN

File: share/eds/eds.go
Comment (Wondertan) [reply share/eds/eds.go:226]: No actionable item here, and just ranting about how this could be better. I will likely touch upon this again during the extraction and either make an issue or fix in place if its an easy fix

**Review by renaynay — APPROVED** [review]
Looks good so far - we should put this on robusta and check it out. When's the tx fuzzer gonna be ready @rootulp @cmwaters ?

**Comment by renaynay** [comment]
Also @rootulp can you run integration tests locally

**Comment by renaynay** [comment]
re https://github.com/celestiaorg/celestia-node/pull/1999#issuecomment-1513909796 @musalbas 

I think that should be a utility inside of celestia-app since it should also reject invalid namespaces, no? 

we can import it.

File: share/eds/testdata/example-root.json
Comment (evan-forbes) [reply share/eds/testdata/example-root.json:3]: > AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH

**Comment by rootulp** [comment]
> Also @rootulp can you run integration tests locally

```
$ make test-swamp
--> Running swamp tests
# github.com/rjeczalik/notify
cgo-gcc-prolog:217:2: warning: 'FSEventStreamScheduleWithRunLoop' is deprecated: first deprecated in macOS 13.0 - Use FSEventStreamSetDispatchQueue instead. [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/FSEvents.framework/Headers/FSEvents.h:1138:1: note: 'FSEventStreamScheduleWithRunLoop' has been explicitly marked deprecated here
ok  	github.com/celestiaorg/celestia-node/nodebuilder/tests	87.951s
```

celestia-app does reject invalid namespaces. I'm not sure yet if this validation is performed when interacting with the celestia-node RPC but will check.

**Comment by rootulp** [comment]
> Likely when we will go through the upgrade process during ITN

FWIW the versioned namespaces breaking change in celestia-app wasn't written with backwards compatability in-mind. Therefore, deploying a celestia-app or celestia-node release with this change on ITN will require additional investigation / work. My biggest concern is that nodes of the NMT are larger after this change and it's not clear what breaks before/after that change.

If we need to support deploying this to ITN then we can start working on that but my impression was that it was safe to make this change (versioned namespaces) + a few other in-flight changes in celestia-app in a backwards incompatible way.

**Comment by musalbas** [comment]
I don't think it makes sense to include this in the BSR upgrade and to add backwards compatibility logic just for that.

**Comment by rootulp** [comment]
FWIW I tried cherry-picking https://github.com/celestiaorg/celestia-app/commit/db1ccfdc800bddfbcbeaa635c8d36de2df6723f7 to the celestia-app `0.14.x` release branch but it depends on a few other changes (i.e. at least https://github.com/celestiaorg/celestia-app/commit/bba1852d7ff57228e98d100a92ae4e389a343d08 and https://github.com/celestiaorg/celestia-app/commit/058442df522867a23e333977783a1d8147a924f1). If @evan-forbes doesn't have objections, I'm more inclined to cut `0.15.x-rc0` from current celestia-app `main` and then bump to that in this PR.

**Comment by Wondertan** [comment]
@rootulp, we are good with it, would be awesome to include latest nmt changes there as well

**Comment by rootulp** [comment]
All tests pass locally with the exception of

```
--- FAIL: TestListenerWithNonEmptyBlocks (12.36s)
    testing.go:1446: race detected during execution of test
```

but that appears expected per [this](https://github.com/celestiaorg/celestia-node/actions/runs/4870601950/jobs/8686551880#step:4:1107) failing CI job on `main`

**Comment by rootulp** [comment]
I think there is a bug in header/store initialization 

When I start celestia-node the first time

```
$ celestia bridge start --p2p.network private --core.ip 127.0.0.1 --gateway
2023-05-04T13:46:55.006-0400	INFO	node	nodebuilder/module.go:22	Accessing keyring...
2023-05-04T13:46:55.010-0400	INFO	module/state	state/keyring.go:40	constructed keyring signer	{"backend": "test", "path": "/Users/rootulp/.celestia-bridge-private/keys", "key name": "my_celes_key", "chain-id": "private"}
2023-05-04T13:46:55.014-0400	WARN	module/gateway	gateway/module.go:25	Gateway is enabled, however gateway endpoints are not authenticated. Use with caution!
2023-05-04T13:46:55.044-0400	INFO	badger	v2@v2.2007.4/levels.go:183	All 0 tables opened in 0s

2023-05-04T13:46:55.079-0400	INFO	fraudserv	fraudserv/helpers.go:29	joining topic	{"id": "/private/fraud-sub/badencoding/v0.0.1"}
2023-05-04T13:46:55.080-0400	INFO	fraudserv	fraudserv/service.go:97	starting fraud proof service	{"protocol ID": "/private/fraud/v0.0.1"}
2023-05-04T13:46:55.083-0400	INFO	header/store	store/store.go:120	initialized head	{"height": 10, "hash": "EEDD25F4B935F52A5B11A077B2CCE31D55DB1A6A0D4EB21A25F59994FA07C307"}
2023-05-04T13:46:55.083-0400	FATAL	header/store	store/heightsub.go:87	PLEASE FILE A BUG REPORT: headers given to the heightSub are in the wrong order
```

When I start it again, it starts successfully. No other changes made in-between starts:

```
$ celestia bridge start --p2p.network private --core.ip 127.0.0.1 --gateway
2023-05-04T13:47:03.425-0400	INFO	node	nodebuilder/module.go:22	Accessing keyring...
2023-05-04T13:47:03.428-0400	INFO	module/state	state/keyring.go:40	constructed keyring signer	{"backend": "test", "path": "/Users/rootulp/.celestia-bridge-private/keys", "key name": "my_celes_key", "chain-id": "private"}
2023-05-04T13:47:03.429-0400	WARN	module/gateway	gateway/module.go:25	Gateway is enabled, however gateway endpoints are not authenticated. Use with caution!
2023-05-04T13:47:03.432-0400	INFO	badger	v2@v2.2007.4/levels.go:183	All 0 tables opened in 0s

2023-05-04T13:47:03.437-0400	INFO	badger	v2@v2.2007.4/value.go:1158	Replaying file id: 0 at offset: 0

2023-05-04T13:47:03.437-0400	INFO	badger	v2@v2.2007.4/value.go:1178	Replay took: 19.583µs

2023-05-04T13:47:03.446-0400	INFO	fraudserv	fraudserv/helpers.go:29	joining topic	{"id": "/private/fraud-sub/badencoding/v0.0.1"}
2023-05-04T13:47:03.446-0400	INFO	fraudserv	fraudserv/service.go:97	starting fraud proof service	{"protocol ID": "/private/fraud/v0.0.1"}
2023-05-04T13:47:03.447-0400	INFO	header/store	store/store.go:169	loaded head	{"height": 10, "hash": "EEDD25F4B935F52A5B11A077B2CCE31D55DB1A6A0D4EB21A25F59994FA07C307"}
2023-05-04T13:47:03.447-0400	INFO	header/p2p	p2p/subscriber.go:40	joining topic	{"topic ID": "/private/header-sub/v0.0.1"}
2023-05-04T13:47:03.450-0400	INFO	header/sync	sync/sync_head.go:116	new network head	{"height": 14, "hash": "AE7558FA88D9D39648D7664A2106D1B527E6EF6693AB916BB42803417FD256E8"}
2023-05-04T13:47:03.450-0400	INFO	header/sync	sync/sync.go:200	syncing headers	{"from": 11, "to": 14}
2023-05-04T13:47:03.450-0400	INFO	header/store	store/store.go:353	new head	{"height": 13, "hash": "B96289B6369C25F39BE6EAD2ED045AE6AE10E6FC0767922BBFDBB9B85B9C2798"}
2023-05-04T13:47:03.450-0400	INFO	header/store	store/store.go:353	new head	{"height": 14, "hash": "AE7558FA88D9D39648D7664A2106D1B527E6EF6693AB916BB42803417FD256E8"}
2023-05-04T13:47:03.450-0400	INFO	header/sync	sync/sync.go:218	finished syncing headers	{"from": 11, "to": 14, "elapsed time": 0.000831458}
2023-05-04T13:47:03.451-0400	INFO	header/p2p	p2p/server.go:65	server: listening for inbound header requests	{"protocol ID": "/private/header-ex/v0.0.3"}
2023-05-04T13:47:03.451-0400	INFO	rpc	rpc/server.go:96	server started	{"listening on": "0.0.0.0:26658"}
2023-05-04T13:47:03.451-0400	INFO	gateway	gateway/server.go:51	server started	{"listening on": "0.0.0.0:26659"}
2023-05-04T13:47:03.451-0400	INFO	node	nodebuilder/node.go:108
```

Tagging @Wondertan for awareness

**Comment by evan-forbes** [comment]
what's the status of this? do we need to bump to something even more recent lol

**Comment by rootulp** [comment]
```
$ make test-swamp
--> Running swamp tests
ok  	github.com/celestiaorg/celestia-node/nodebuilder/tests	64.908s
```

It passes integ tests locally so I think we just need @celestiaorg/celestia-node to review 🙏

**Comment by Wondertan** [comment]
> are we wanting to bump main to a significantly different version of the app than what BSR is running?

@evan-forbes, I don't think so; we can keep this open until we decide to break main compatibility with BSR

**Comment by rootulp** [comment]
celestia-node will upgrade directly to celestia-app v1.0.0-rc0 when that is available so this PR doesn't need to remain open.

**Pull request closed without merging**
